paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -224,14 +224,16 @@ internal class DefaultFlowController @Inject internal constructor(
     }
 
     override fun presentPaymentOptions() {
-        val state = runCatching {
-            requireNotNull(viewModel.state)
-        }.getOrElse {
-            error(
-                "FlowController must be successfully initialized using " +
-                    "configureWithPaymentIntent() or configureWithSetupIntent() " +
-                    "before calling presentPaymentOptions()"
-            )
+        val state = viewModel.state ?: error(
+            "FlowController must be successfully initialized " +
+                "using configureWithPaymentIntent(), configureWithSetupIntent() or " +
+                "configureWithIntentConfiguration() before calling presentPaymentOptions()."
+        )
+
+        if (configurationHandler.isConfiguring || viewModel.didLastConfigurationFail) {
+            // "Cannot call presentPaymentOptions when the last update call
+            // has not yet finished or failed."
+            return
```

Comment:
This would be a good place for a Log, or probably better yet an event from a (non existent) event listener.